### PR TITLE
Catch Northstar errors.

### PR DESCRIPTION
**Pull request by DFurnes**
#### Changes

When sending users to Northstar, we may encounter errors. Instead of crashing and showing a "Whoops!", we should be prepared to handle each type of error:

A validation may occur if we try to "upsert" a conflict, like if I registered with my real phone number and a `dfurnes+1234@dosomething.org` email, even though that phone number is already registered to a `dfurnes@dosomething.org` account. (See DoSomething/northstar#346.)

We may also get an `APIException` or a `ConnectionException` if another unexpected failure occurs. These are totally not good, but we should still record the vote and user in the local database, and trigger some monitoring alerts so we can fix it ASAP! 👷
#### How should this be reviewed?

You can probably just skim the diff. 👀 

---

For review: @katiecrane @weerd

### CODE REVIEW

File: app/Http/Controllers/VotesController.php [hunk 1]
```diff
@@ -72,7 +72,14 @@ public function store(VoteRequest $request)
             event(new UserCastFirstVote($vote));
         }
 
-        $northstarUser = Northstar::updateUser($user, $vote->candidate);
+        // If we have a valid Northstar ID, let's try to update their profile with their vote.
+        if ($user->northstar_id !== 'CONFLICT' && $user->northstar_id !== 'ERROR_CONNECTION') {
```
Comment (katiecrane) [inline app/Http/Controllers/VotesController.php]: do we not hit this if the northstar id is ERROR?

Comment (DFurnes) [inline app/Http/Controllers/VotesController.php]: Oh doy. Yeah that's a mistake!

Comment (DFurnes) [inline app/Http/Controllers/VotesController.php]: Fixed! 🐛 🔨

**Comment by katiecrane** [comment]
only thing I noticed is fixed now 😄 ⛵

**Comment by DFurnes** [comment]
🎉

**Pull request merged**